src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -912,13 +939,57 @@ export default class DataTable extends Vue {
     this.slicedData = this.sliceData(this._sortedData || this._serializedDataBody);
   }
 
+  _showSelectedRows(isSelected: boolean) {
+    /* We are not supporting nested rows show selection for now. This code is to handle a bug specific to
+    nested row selection which needs to be modified/refactored in future */
+    const nestedData = this.slicedData.find(v => 'nestedContent' in v && 'table' in v.nestedContent);
+    const nonNestedRows = this.slicedData.filter(v => v.rowId != nestedData?.rowId);
+    const nestedRowIds = nonNestedRows.filter(v => this.selectedRows.includes(v.rowId));
+
+    if (nestedRowIds.length > 0 || this.accordionsExpanded.length == 0) {
+      const selectedRows = this._serializedDataBody.filter((row: DataTableRow) => {
+        return this.selectedRows.some(rowId => rowId === row.rowId);
+      });
+      /******/
+
+      this.activePage = 1;
+      this.slicedData = this.sliceData(selectedRows);
+
+      if (this.selectedRows.length > 10) {
+        const pageChangeEventData: DataTablePageChange = {
+          page: 1,
+        };
+        pageChangeEventData.data = this.slicedData;
+        this._checkSelectAllCheckbox();
+        this.$emit(PAGINATION_EVENTS.PAGE_CHANGE, pageChangeEventData);
+      }
```

Comment:
@janibolkvadze This is needed to handle a scenario where we select multiple checkboxes in multiple pages. When we select show selected only, it displays only the selected rows as expected with active page being 1.